cmake_flags_generator.py

```diff
@@ -0,0 +1,138 @@
+import re
+import os.path
+from typing import TextIO, List, Tuple, Optional, Dict
+
+Entity = Tuple[str, str, str]
+
+# https://regex101.com/r/R6iogw/11
+cmake_option_regex: str = r"^\s*option\s*\(([A-Z_0-9${}]+)\s*(?:\"((?:.|\n)*?)\")?\s*(.*)?\).*$"
+
+output_file_name: str = "cmake_flags_and_output.md"
```

Comment:
It is strange that we put these files in root directory.
Maybe move it to `cmake`?

Reply:
I'll change the location of these files today.